packages/webviz-core/src/panels/ThreeDimensionalViz/TopicSettingsEditor/LaserScanSettingsEditor.js

```diff
@@ -6,19 +6,41 @@
 //  found in the LICENSE file in the root directory of this source tree.
 //  You may not use this file except in compliance with the License.
 
-import React from "react";
+import React, { useCallback } from "react";
+import { type Color } from "regl-worldview";
 
 import { CommonPointSettings, CommonDecaySettings, type TopicSettingsEditorProps } from ".";
+import ColorPickerForTopicSettings from "./ColorPickerForTopicSettings";
+import { SLabel } from "./common";
 import Flex from "webviz-core/src/components/Flex";
 import type { LaserScan } from "webviz-core/src/types/Messages";
 
+export type ColorMode = { mode: "flat", flatColor: Color };
+
+export const DEFAULT_FLAT_COLOR = { r: 0.5, g: 0.5, b: 1, a: 1 };
+
 type LaserScanSettings = {|
   pointSize?: ?number,
   pointShape?: ?string,
   decayTime?: ?number,
+  colorMode: ?ColorMode,
 |};
+
 export default function LaserScanSettingsEditor(props: TopicSettingsEditorProps<LaserScan, LaserScanSettings>) {
-  const { settings, onFieldChange } = props;
+  const { settings, onFieldChange, onSettingsChange } = props;
+
+  const onColorModeChange = useCallback(
```

Comment:
Maybe let's keep it simple by only allowing flat color for now unless you have a strong use case for rainbow/gradient colors? Also you'd need to add SegmentControl (for selecting the color modes) and Select (to choose which value field to color from) as well ([example](https://github.com/cruise-automation/webviz/blob/master/packages/webviz-core/src/panels/ThreeDimensionalViz/TopicSettingsEditor/PointCloudSettingsEditor.js#L104-L153)). 

You can refer[ general marker settings](https://github.com/cruise-automation/webviz/blob/master/packages/webviz-core/src/panels/ThreeDimensionalViz/TopicSettingsEditor/MarkerSettingsEditor.js#L32-L34) for the color picker UI.

Reply:
Yeah, I tried to keep the color settings identical to the pointcloud color settings. But agree it adds a lot of complexity. After reading this comment I took another look at it, the actual handling of the color modes is also a [lot of code](https://github.com/cruise-automation/webviz/blob/725912f321248aa8efab8c2bbad06edd2a86712f/packages/webviz-core/src/panels/ThreeDimensionalViz/commands/PointClouds/index.js). If the LaserScan is ever to support gradients for the intensity, it may actually be better to have a small converter that just does the Ray -> Point transform and then use the topic settings and render backend that's in use for the PointCloud. Because the PointCloud2 display is a superset of the features a LaserScan would need.

For now, I decided to follow your advice and simplify it into one flat color.